com.unity.formats.fbx/Editor/ConvertToNestedPrefab.cs

```diff
@@ -786,8 +847,38 @@ internal static void UpdateFromSourceRecursive(GameObject dest, GameObject sourc
                     PrefabUtility.UnpackPrefabInstance(sourceGO, PrefabUnpackMode.Completely, InteractionMode.AutomatedAction);
                 }
 
-                FixSceneReferences(sourceGO, destGO, source);
-                CopyComponents(destGO, sourceGO, source, goDict);
+                // Map of objects that may need to have their scene references fixed.
+                // Maps from original to new object.
+                var fixSceneRefsMap = new Dictionary<Object, Object>()
+                {
+                    { sourceGO, destGO }
+                };
+
+                CopyComponents(destGO, sourceGO, source, fixSceneRefsMap);
+
+                // Fix references of other objects in the scene
+#if UNITY_2021_2_OR_NEWER
+                if(info.sceneObjectsWithReference != null)
+                {
+                    var items = info.sceneObjectsWithReference;
+                    foreach (var item in items)
+                    {
+                        FixSceneReferenceToObject(item, source, fixSceneRefsMap);
+                    }
+                }
+#else // UNITY_2021_2_OR_NEWER
+                foreach (var obj in fixSceneRefsMap.Keys)
+                {
+                    // find and fix the references in the scene
+                    var sceneObjs = GetSceneReferencesToObject(obj);
```

Comment:
can you do this earlier and stuff it into info.sceneObjectsWithReference, then remove the #if here?